src/appleseed.shaders/src/appleseed/as_subsurface.osl

```diff
@@ -315,74 +343,89 @@ shader as_subsurface
 
     normal Nn = normalize(in_bump_normal);
 
-    if (in_sss_amount * max(in_color) > 0.0)
-    {
-        color albedo = clamp(in_sss_amount * in_color, 0.01, 0.99);
+    color albedo = clamp(in_sss_amount * in_color, 0.01, 0.99);
 
-        if (sss_profile == "randomwalk")
+    if (sss_profile == "randomwalk")
+    {
+        if (in_specular_weight > 0.0)
         {
-            out_outColor += as_subsurface(
-                sss_profile,
-                Nn,
-                albedo,
-                in_sss_mfp_scale * in_sss_mfp,
-                in_ior,
-                "fresnel_weight", in_fresnel_weight);
+            out_outColor += in_specular_weight *
+                as_randomwalk_glass(
+                    Nn,
+                    albedo,
+                    in_sss_mfp_scale * in_sss_mfp,
+                    in_ior,
+                    in_specular_roughness,
+                    "fresnel_weight", 1.0,
+                    "volume_anisotropy", in_volume_anisotropy);
         }
-        else
+        if (in_specular_weight < 1.0)
```

Comment:
I'm not sure to understand the logic. You're (nearly) always computing and adding together glass and diffuse random-walk components? In other words, do you confirm that the absence of `else` is on purpose here?

Reply:
Still WIP but the idea is doing a linear blend between the *diffuse* and *glass* randomwalk profiles via *specular weight*. The GGX term is not added yet, the *glass* random walk is not accounting for reflectance.It would give some measure of control for *randomwalk* while sharing some parameters with the specular term.
Alternatively we could add *diffuse randomwalk* and *glass randomwalk*, i'm open to suggestions.

Reply:
Regarding breaking changes, i know in Maya and Gaffer adding a parameter poses no problem, but in Max adding a parameter needs to be accounted for, correct @usakhelo ?
In blender i'm not sure, @jdent02 ?

Reply:
Yes, that is unfortunately correct. I tried it out with your new version of asSubsurface  and it breaks scenes (Error loading ParamBlock2) which have the current version of asSubsurface as a material assigned to objects.

Reply:
I'll leave the *glassy* random walk profile and phase function control to a v2 version but retain the bounces changes which should pose no problem.

Reply:
Removed the glassy random walk profile.